File: chromedp/stealth-mode/stealth-mode.go
package main

import (
	"context"
	"fmt"
	"io/ioutil"
	"log"

	"github.com/chromedp/chromedp"
)

// Check your API key and web socket endpoint from 
// My Products (https://browseasy.com/products/) page.

// Assuming that it's stored in the environment variables, 
// e.g. wss://freemium.browseasy.com
var BROWSEASY_ENDPOINT = os.Getenv("BROWSEASY_ENDPOINT")

// Assuming that it's stored in the environment variables
// e.g. ccc70169f82f4c7c8a33ecca21c1becf
var BROWSEASY_API_KEY = os.Getenv("BROWSEASY_API_KEY")

// Your unique connection string.
var BROWSEASY_CONNECTION_STRING = fmt.Sprintf("%s?code=%s", BROWSEASY_ENDPOINT, BROWSEASY_API_KEY)

// Your unique connection string with stealth mode enabled.
var BROWSEASY_STEALTH_CONNECTION_STRING = fmt.Sprintf("%s&stealth", BROWSEASY_CONNECTION_STRING)

func main() {
	// create allocator context for use with creating a browser context later
	allocatorContext, cancel := chromedp.NewRemoteAllocator(context.Background(), BROWSEASY_STEALTH_CONNECTION_STRING)
	defer cancel()

	// create context
	ctxt, cancel := chromedp.NewContext(allocatorContext)
	defer cancel()

	var buf []byte
	// capture a screenshot
	if err := chromedp.Run(ctxt,
		chromedp.Navigate("https://fingerprintjs.github.io/fingerprintjs/"),
		chromedp.CaptureScreenshot(&buf),
	); err != nil {
		log.Fatalf("Failed: %v", err)
	}
	if err := ioutil.WriteFile("fingerprintjs.png", buf, 0o644); err != nil {
		log.Fatal(err)
	}
}

File: playwright/Java/stealth-mode/src/main/java/com/browseasy/example/StealthMode.java
package com.browseasy.example;

import java.nio.file.Paths;

import com.microsoft.playwright.*;
import com.microsoft.playwright.Browser.NewContextOptions;

public class StealthMode {
  // Update your endpoint with the one dedicated to your own plan or use freemium.
  static final String BROWSEASY_ENDPOINT = "wss://freemium.browseasy.com";

  // Check your API key from My Products page for your plan and update accordingly.
  static final String BROWSEASY_API_KEY = "YOUR-API-KEY";

  // Your unique connection string.
  static final String BROWSEASY_CONNECTION_STRING = "%s?code=%s".formatted(BROWSEASY_ENDPOINT, BROWSEASY_API_KEY);

  // Your unique connection string with stealth mode enabled.
  static final String BROWSEASY_STEALTH_CONNECTION_STRING = "%s&stealth".formatted(BROWSEASY_CONNECTION_STRING);
  public static void main(String[] args) {
    try (Playwright playwright = Playwright.create()) {
      // Run headless browser locally, no more.
      // Browser browser = playwright.chromium().launch();
      
      // Run headless browser on the cloud.
      Browser browser = playwright.chromium().connectOverCDP(BROWSEASY_STEALTH_CONNECTION_STRING);

      // Unfortunately, default context is always overwritten by playwright.

      // We need to create a new context, so that Browseasy can handle
      // screen resolution and viewport adjustment.

      // When stealth mode enabled, Browseasy will launch your browser 
      // with a random resolution based on a sample distribution. 
      // See https://gs.statcounter.com/screen-resolution-stats/desktop/worldwide
      BrowserContext context = browser.newContext(new NewContextOptions().setViewportSize(null));

      Page page = context.newPage();
      page.navigate("https://browseasy.com/");

      page.screenshot(new Page.ScreenshotOptions().setPath(Paths.get("browseasy.png")));

      browser.close();
    }
  }
}